sympy/tensor/tensor.py

```diff
@@ -1771,7 +1774,21 @@ def __call__(self, *indices, **kw_args):
         A(a, -b)
 
         """
-        tensor = Tensor(self, indices, **kw_args)
+
+        updated_indices = []
+        for idx, typ in zip(indices, self.index_types):
+            if isinstance(idx, str):
```

Comment:
Ideally, we should do some string processing, for example `idx = idx.strip().replace(" ", "")` to remove possible whitespaces.

Reply:
I will think about other possibly useful processing and make an update.

Reply:
I have made the update.